test/extensions/filters/common/rbac/engine_impl_test.cc

```diff
@@ -27,25 +27,20 @@ void checkEngine(const RBAC::RoleBasedAccessControlEngineImpl& engine, bool expe
 }
 
 TEST(RoleBasedAccessControlEngineImpl, Disabled) {
-  envoy::config::filter::http::rbac::v2::RBACPerRoute config;
-  config.set_disabled(true);
-  checkEngine(RBAC::RoleBasedAccessControlEngineImpl(config), true);
-  checkEngine(
-      RBAC::RoleBasedAccessControlEngineImpl(envoy::config::filter::http::rbac::v2::RBAC(), true),
-      true);
+  checkEngine(RBAC::RoleBasedAccessControlEngineImpl(
```

Comment:
Since we no longer have `disabled` in proto, you should rename this to `Empty`.

This expectation is wrong. The default action is ALLOW, so per https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/rbac/v2alpha/rbac.proto#L51 , it should deny the rest (everything).

Please remove the special handling with variable `engine_disabled_` above, less special cases, less confusing...

Reply:
That is also incorrect. When disabled, the filter is not supposed to evaluate at all and act as if the filter is not present (ie, always allow). This is partly why it was included explicitly for per-route configs to make that distinction clear that there is no logical behavior to expect from the rest of the filter.

Reply:
moved checking rbac enabled logic to filter, also filter.decodeHeaders won't enter engine if rbac is disabled.